tools/sof-fw-boot-test.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+"""Module to stress test firmware boot"""
+
+import argparse
+import logging
+import sys
+import time
+
+# define command line arguments
+def parse_cmdline():
+	"""Function to parse the command line arguments"""
+	parser = argparse.ArgumentParser(add_help=True, formatter_class=argparse.RawTextHelpFormatter,
+									 description='A script for stress testing firmware boot')
+	parser.add_argument('-i', '--iter', type=int, default=100,
+						help="number of firmware boot iterations")
+	parser.add_argument('-f', '--firmware', type=str, required=True, help='firmware filename')
+	parser.add_argument('-p', '--fw_path', type=str, required=True,
+						help='path to the firmware file relative to /lib/firmware')
+	return vars(parser.parse_args())
+
+def main():
+	"""Main function for stress testing"""
+	logging.basicConfig(level=logging.INFO, format='%(message)s')
+	cmd_args = parse_cmdline()
+
+	# Get firmware file path
+	fw_path = cmd_args['fw_path']
+
+	# Get firmware file name
+	fw_filename = cmd_args['firmware']
+
+	num_iter = cmd_args['iter']
+	output = f"""==============================================================================
+	Starting boot stress test with:
+	Firmware filename: {fw_filename}
+	Path to firmware file: {fw_path}
+	Number of Iterations: {num_iter}
+=============================================================================="""
+	logging.info(output)
+
+	# set path to the depbugfs entries
+	debugfs_path = '/sys/kernel/debug/sof/dsp_ops'
+
+	# set the firmware filename & path
+	with open(debugfs_path + '/fw_filename', 'w', encoding="utf8") as dbgf:
+		dbgf.write(fw_filename + "\n")
+	with open(debugfs_path + '/fw_path', 'w', encoding="utf8") as dbgf:
+		dbgf.write(fw_path + "\n")
+
+	total_boot_time_ms = 0
+	min_boot_time_ms = sys.maxsize
+	max_boot_time_ms = 0
+
+	for i in range(num_iter):
```

Comment:
Can you create some new, `power_cycle()`, one-iteration function?

This solves 3 problems:

- `main()` is getting too big
- Drops a LOT of whitespace
- Makes interactive testing much easier because know you can interactively `import sof-fw-boot-test.py` and invoke `power_cycle()`. Try it!

Reply:
done